src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -724,11 +742,40 @@ async def do_update(self, id, data):
     async def do_delete(self, id):
         """
         Delete a domain by id. Deletion of default system domains is not permitted.
+        In case of registry config for clustered server, this will remove all smb4.conf
+        entries for the domain associated with the id.
         """
+        await self.middleware.call("smb.cluster_check")
         if id <= 5:
             entry = await self._get_instance(id)
             raise CallError(f'Deleting system idmap domain [{entry["name"]}] is not permitted.', errno.EPERM)
-        await self.middleware.call("datastore.delete", self._config.datastore, id)
+
+        ha_mode = SMBHAMODE[(await self.middleware.call('smb.get_smb_ha_mode'))]
+        if ha_mode != SMBHAMODE.CLUSTERED:
+            return await self.middleware.call("datastore.delete", self._config.datastore, id)
+
+        return await self.reg_delete(id)
+
+    @filterable
+    async def query(self, filters=None, options=None):
```

Comment:
This will cause the following error: `ValueError: All public method default arguments should be specified in @accepts()`

```suggestion
    async def query(self, filters, options):
```